### PR TITLE
tso: add some metrics for Keyspace Group operations

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref  #7011.

### What is changed and how does it work?

```commit-message
Add some metrics for Keyspace Group operations.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/tso/keyspace_group_manager.go [hunk 1]
```diff
@@ -858,10 +866,24 @@ func (kgm *KeyspaceGroupManager) updateKeyspaceGroupMembership(
...
+		if oldGroup.IsSplitTarget() && !newGroup.IsSplitting() {
+			kgm.ams[groupID].GetMember().(*member.Participant).SetCampaignChecker(nil)
+			splitTime := kgm.splittingGroups[groupID]
+			delete(kgm.splittingGroups, groupID)
+			kgm.metrics.splitTargetGauge.Desc()
+			kgm.metrics.splitDuration.Observe(time.Since(splitTime).Seconds())
+		}
+		// SplitSource -> !SplitSource
+		if oldGroup.IsSplitSource() && !newGroup.IsSplitting() {
+			kgm.metrics.splitSourceGauge.Desc()
```
Comment (rleungx) [inline pkg/tso/keyspace_group_manager.go:881]: Dec?

Comment (JmPotato) [reply pkg/tso/keyspace_group_manager.go]: Fixed.

**Review by rleungx — APPROVED** [review]

File: pkg/tso/keyspace_group_manager.go [hunk 2]
```diff
@@ -72,14 +72,14 @@ type state struct {
 	// keyspaceLookupTable is a map from keyspace to the keyspace group to which it belongs.
 	keyspaceLookupTable map[uint32]uint32
 	// splittingGroups is the cache of splitting keyspace group related information.
-	splittingGroups map[uint32]struct{}
+	splittingGroups map[uint32]time.Time
```
Comment (lhy1024) [inline pkg/tso/keyspace_group_manager.go:75]: How about adding a comment to indicate what it stores?

Comment (JmPotato) [reply pkg/tso/keyspace_group_manager.go:75]: Addede.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**